cloud/ec2_client.go

```diff
@@ -944,14 +944,14 @@ func (c *awsClientImpl) AssumeRole(ctx context.Context, input *sts.AssumeRoleInp
 			msg := makeAWSLogMessage("AssumeRole", fmt.Sprintf("%T", c), input)
 			output, err = c.stsClient.AssumeRole(ctx, input)
 			if err != nil {
+				grip.Debug(message.WrapError(err, msg))
```

Comment:
Could it be worth doing this only at the end so that we aren't logging for every retry, if it does eventually work?